gyp/pylib/gyp/MSVSNew.py

```diff
@@ -4,22 +4,17 @@
 
 """New implementation of Visual Studio project generation."""
 
+import hashlib
 import os
 import random
 
 import gyp.common
 
-# hashlib is supplied as of Python 2.5 as the replacement interface for md5
-# and other secure hashes.  In 2.6, md5 is deprecated.  Import hashlib if
-# available, avoiding a deprecation warning under 2.6.  Import md5 otherwise,
-# preserving 2.4 compatibility.
 try:
-  import hashlib
-  _new_md5 = hashlib.md5
-except ImportError:
-  import md5
-  _new_md5 = md5.new
-
+    cmp             # Python 2
+except NameError:
+    def cmp(x, y):  # Python 3
+        return (x > y) - (x < y)
```

Comment:
Since `cmp` is only used in one place in this file, I'd just open-code it there. Easier to understand and maintain than version-specific branching.